egs2/aishell4/asr1/local/data.sh

```diff
@@ -0,0 +1,359 @@
+    log "selecting lines for train, dev and test ..."
+
+    utils/subset_data_dir.sh --first data/train 1000 data/dev
+    n=$(($(wc -l < data/train/text) - 1000))
+    utils/subset_data_dir.sh --last data/train ${n} data/train_nodev
+
+fi
+
+
+if [ ${stage} -le 10 ] && [ ${stop_stage} -ge 10 ] ; then 
+    log "resorting the files ..."
+    log "train ..."
+    sort data/train_nodev/utt2spk -o data/train_nodev/utt2spk
+    utils/utt2spk_to_spk2utt.pl data/train_nodev/utt2spk > data/train_nodev/spk2utt
+    sort data/train_nodev/wav.scp -o data/train_nodev/wav.scp
+    sort data/train_nodev/text -o data/train_nodev/text
+    log "files sorted"
+    log "dev ..."
+    sort data/dev/utt2spk -o data/dev/utt2spk
+    utils/utt2spk_to_spk2utt.pl data/dev/utt2spk > data/dev/spk2utt
+    sort data/dev/wav.scp -o data/dev/wav.scp
+    sort data/dev/text -o data/dev/text
+    log "files sorted"
+
+
+fi 
+
+
+
+
+
+
+#################################################################
+#####      Combining with aishell1 data  (train only for now)   
+#################################################################
+
+
+# pay attention : sorting issues with utt2spk :  (fix this by making speaker-ids prefixes of utt-ids)
+
+if [ ${stage} -le 11 ] && [ ${stop_stage} -ge 11 ] ; then 
+    
+    aishell1_data=/ocean/projects/cis210027p/berrebbi/espnet/egs2/aishell/asr1/data/train
+    aishell4_data=data/train_nodev
+
+    u2s=/ocean/projects/cis210027p/berrebbi/espnet/egs2/aishell/asr1/data/train/utt2spk
+    awk 'BEGIN {FS=" "; OFS="\n"}; {print $1" "$1}' $u2s > /ocean/projects/cis210027p/berrebbi/espnet/egs2/aishell/asr1/data/train/utt2spk2
+
+    mv /ocean/projects/cis210027p/berrebbi/espnet/egs2/aishell/asr1/data/train/utt2spk2 /ocean/projects/cis210027p/berrebbi/espnet/egs2/aishell/asr1/data/train/utt2spk
+    
+    utils/combine_data.sh data/combined_aishell_dir/train $aishell1_data $aishell4_data 
+
+    
+    sort data/combined_aishell_dir/train/utt2spk -o data/combined_aishell_dir/train/utt2spk
+    utils/utt2spk_to_spk2utt.pl data/combined_aishell_dir/train/utt2spk > data/combined_aishell_dir/train/spk2utt
+    sort data/combined_aishell_dir/train/wav.scp -o data/combined_aishell_dir/train/wav.scp
+    sort data/combined_aishell_dir/train/text -o data/combined_aishell_dir/train/text
+
+    wc -l data/combined_aishell_dir/train/*
+
+
+fi 
+
+
+
+
+
+##########################
+##       test set 
+##########################
+
+
+wav_list_aishell4=${AISHELL4}/test/wav_list.txt
+text_grid_aishell4=${AISHELL4}/test/TextGrid_list.txt
+
+output_folder=$PWD/data
+
+if [ ${stage} -le 12 ] && [ ${stop_stage} -ge 12 ] ; then 
+
+    log "generating asr training data ..."
+    log "(this can take some time)"
+ 
+    python "$FOLDER"/data_preparation/generate_asr_trainingdata.py  --output_dir "$output_folder"/test --mode train --aishell4_wav_list "$wav_list_aishell4" --textgrid_list "$text_grid_aishell4" || log "ca a pas marché" ;
```

Comment:
Hi, I am reading the AISHELL-4 recipe recently.
I feel a bit confused by this line because it uses `generate_asr_trainingdata.py` to generate the test data, while the [`felixfuyihui/AISHELL-4`](https://github.com/felixfuyihui/AISHELL-4) repository actually provides [data_preparation/generate_nospk_testdata.py](https://github.com/felixfuyihui/AISHELL-4/blob/master/data_preparation/generate_nospk_testdata.py) for creating the speaker-independent test data.

Is this correct?